dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl

```diff
@@ -134,8 +147,14 @@ RUN set -e ; \
   chmod +x /usr/bin/tini
 
 COPY docker-entrypoint /usr/local/bin/docker-entrypoint
+
+# We do not want to add the agent user to the root group and because of backwards compatibility we can do it only for the new Wolfi image
 RUN groupadd --gid 1000 {{ .BeatName }} && \
+{{- if contains .from "wolfi" }}
+    useradd -M --uid 1000 --gid 1000 {{ .user }} && \
```

Comment:
Does this also require the changes here - https://github.com/elastic/elastic-agent/pull/4925

Reply:
I made this change by a request from @cmacknz 
It's better to ask him for the context.

Reply:
@cmacknz Can this change be made without the above PR?

Reply:
We are figuring that out.

What I want is for this container to have the least privileges possible by default, the container version of an unprivileged agent. We can then escalate back to being root when deploying it via the k8s security context, that may require the PR above to work properly. TBD.